src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs

```diff
@@ -149,6 +149,13 @@ public async Task<IActionResult> List(ListContentsViewModel model, PagerParamete
                 {
                     query = query.With<ContentItemIndex>(x => x.ContentType.IsIn(listableTypes));
                 }
+
+                // Exclude non listable content types
```

Comment:
This will break other features (you won't be able to view menus anymore), so you can't do this.

I don't entirely understand the issue (better repro steps might help), but is it related to there being no types available to list when changing a user permissions.

If so, it might be that the `if.Any()` three lines up may need to be changed to be an empty enumerable. Assuming that works.

So try removing the `if.Any()` line and see if that fixes the issue

Reply:
The issue is when you create a non listable content type it's disappear only with administrator users, which is totally wrong

I will try what you suggested before

Reply:
> So try removing the if.Any() line and see if that fixes the issue

Removing the `if` that you point out clear all the results, because the `listableTypes` is empty

Reply:
Maybe you could post some screenshots demonstrating the issue as this just doesn't feel like the right fix, and I can't repro the issue

Reply:
To reproduce the issue:

- Create a content type with `Listable unchecked

- Add new content item for the newly added content type, the item will not display in the content items page, which is the correct behavior

- Create a new user with a `Contributor` role

- Log  out and sign in with the newly added user

- Visit the content items page, you will see the new content item, which is wrong

Reply:
> Removing the if that you point out clear all the results, because the listableTypes is empty

This is correct, and is probably (without trying) the correct fix.

The listable in the ui is based on wether the role can edit the type.

So contributor should list nothing, as they can edit nothing.

Editor can, so it should list correctly.

Reply:
I will try it for `Editor` role, but according to the original issue, the content item(s) show up for non admins roles

Reply:
> So contributor should list nothing, as they can edit nothing. (maybe, maybe not... but there is a bigger issue if they're supposed to be all listed)

Contributors don't view or edit their own content, removing the `if` seems works, but I'm still don't know why `Contributors` doesn't view their content, unless it is by design